### PR TITLE
fix retained view references bug; clean up channel fragment oncreate

### PR DESCRIPTION
**Pull request by carterhudson**
~< Jira issue link, if applicable >~

### Description

We were passing fragments as lifecycle owners for views in a couple places instead of the `viewLifecycleOwner`. The `ChannelsFragment` never gets destroyed, so it retains its `ViewModel`, but it _does_ call `onViewCreated` again, so it calls `viewModel::bindView` each time it comes back on screen. Since Observers were effectively observing forever with a `view` reference, views were being retained forever, and data was being pushed to them. To test it for yourself, check out `develop` and put
```kotlin
ChatLogger.Companion.get("${this::class.java.simpleName}@{${System.identityHashCode(this)}}")
            .logW("-> View@{${System.identityHashCode(view)}}")
```
into the `state.observe(lifecycleOwner) { ... }` block inside `ChannelsViewModel.bindView`. Move back and forth between the `Channel(s)Fragment` and the `ChannelFragment` a few times, and you'll see the same ViewModel instance interacting with _many_ different views after a while.

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
~- [ ] Changelog updated with client-facing changes~
~- [ ] New code is covered by unit tests~
~- [ ] Comparison screenshots added for visual changes~
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt [hunk 1]
```diff
@@ -86,4 +86,8 @@ public class ChannelsViewModelImpl(
     private fun requestMoreChannels() {
         chatDomain.useCases.queryChannelsLoadMore(filter, sort).enqueue()
     }
+
+    override fun onCleared() {
```
Comment (ogkuzmin) [inline stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt:90]: What's the reason for non-customized overriding?

File: stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/channel/ChannelViewModelsFactory.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package io.getstream.chat.sample.feature.channel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.getstream.sdk.chat.viewmodel.ChannelHeaderViewModel
+import com.getstream.sdk.chat.viewmodel.MessageInputViewModel
+import com.getstream.sdk.chat.viewmodel.messages.MessageListViewModel
+
+@Suppress("UNCHECKED_CAST")
+class ChannelViewModelsFactory(private val cid: String) : ViewModelProvider.NewInstanceFactory() {
```
Comment (ogkuzmin) [inline stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/channel/ChannelViewModelsFactory.kt:10]: :+1:

File: stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/channel/ChannelFragment.kt [hunk 3]
```diff
@@ -18,26 +19,41 @@ import kotlinx.android.synthetic.main.fragment_channel.*
...
-                            is MessageListViewModel.State.NavigateUp -> navigateUp()
-                        }
+        val backButtonHandler = {
+            messagesViewModel.onEvent(MessageListViewModel.Event.BackButtonPressed)
+        }
+
+        channelHeaderView.onBackClick = { backButtonHandler() }
+
+        activity?.apply {
+            onBackPressedDispatcher.addCallback(
```
Comment (ogkuzmin) [inline stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/channel/ChannelFragment.kt:44]: Never seen this before. Very cool!

File: stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/channel/ChannelFragment.kt [hunk 4]
```diff
@@ -18,26 +19,41 @@ import kotlinx.android.synthetic.main.fragment_channel.*
...
+                viewLifecycleOwner,
+                object : OnBackPressedCallback(true) {
+                    override fun handleOnBackPressed() {
+                        backButtonHandler()
```
Comment (ogkuzmin) [inline stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/channel/ChannelFragment.kt:48]: Maybe inline this backButtonHandler?

**Review by ogkuzmin — APPROVED** [review]

**Review by JcMinarro — DISMISSED** [review]

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt (hunk 1 above)
Comment (carterhudson) [reply stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt:90]: Was using it for debugging and forgot to remove it!

File: stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/channel/ChannelFragment.kt (hunk 3 above)
Comment (carterhudson) [reply stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/channel/ChannelFragment.kt:44]: New to me, too. I just extracted what was already there to a method.

File: stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/channel/ChannelFragment.kt (hunk 4 above)
Comment (carterhudson) [reply stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/channel/ChannelFragment.kt:48]: I think whoever originally implemented it made it a local reference so they could use it twice for channel header back click & regular back click.

**Review by tschellenbach — APPROVED** [review]

**Review by tschellenbach — APPROVED** [review]

**Pull request merged**